File: Codes/gracekoo/27_remove-element.py
# -*- coding: utf-8 -*-
# @Time: 2020/1/3 11:41 下午
# @Author: GraceKoo
# @File: 27_remove-element.py
# @Desc:https://leetcode-cn.com/problems/remove-element/


class Solution:
    def removeElement(self, nums, val: int):
        if not nums:
            return 0
        i = 0
        for j in range(0, len(nums)):
            if nums[j] != val:
                nums[i] = nums[j]
                i += 1
        return i


so = Solution()
print(so.removeElement([0, 1, 2, 2, 3, 0, 4, 2], 2))

File: Codes/yore/27_remove_element/27_remove_element.java
class Solution {
    public int removeElement(int[] nums, int val) {
         int length = nums.length;
        int len = 0;
        for (int i = 0; i < length; i++) {
            if (nums[i] != val) {
                nums[len++] = nums[i];
            }
        }
        return len;
    }
}